cmd/minishift/cmd/start.go

```diff
@@ -127,6 +128,12 @@ func runStart(cmd *cobra.Command, args []string) {
 	setOcProxy()
 	setShellProxy()
 
+	if viper.IsSet("http-proxy") {
+		proxyUrl = viper.GetString(httpProxy)
```

Comment:
-1 for putting proxy information with username and password as their uses are different. As I mentioned username and password is only for Red Hat subscription manager. But proxy is used by docker, OpenShift as well as subscription manager.